src/v/cluster/security_frontend.cc

```diff
@@ -117,6 +118,56 @@ ss::future<std::error_code> security_frontend::update_user(
     return replicate_and_wait(_stm, _as, std::move(cmd), tout);
 }
 
+ss::future<std::error_code> security_frontend::create_role(
+  security::role_name name,
+  security::role role,
+  model::timeout_clock::time_point tout) {
+    if (!_features.local().is_active(
+          features::feature::role_based_access_control)) {
+        vlog(clusterlog.warn, "RBAC feature is not yet active");
+        co_return cluster::errc::feature_disabled;
+    }
+    create_role_cmd cmd(std::move(name), std::move(role));
+    co_return co_await replicate_and_wait(_stm, _as, std::move(cmd), tout);
```

Comment:
These commands seem to assume that you're dispatching them onto the controller leader (unlike the acl CRUD). Is that true this will always be invoked on the controller leader node?

Reply:
Correct. More similar to user CRUD than ACL crud. The call site will always be in an Admin API handler, where we have convenience functions for redirecting _http_ requests to the controller leader ahead of any parsing or controller queries. I'll add comments to that effect in the header.

Reply:
Thanks for the comments!

Reply:
done